### PR TITLE
Cloudsql setup

### PR DESCRIPTION
**Pull request by wshin387**
## Link to Issue Number
[Github supports linking PR's to issues to auto close them!](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue)

#25 

## Description
Set up cloud SQL database.
Added API GET+POST+DELETE endpoints for restaurants db.

## Checklist
- [ x ] All feature request A/C have been met OR the bug has been fixed
- [ x ] I ran all nessisary tests and they pass
- [ x ] I rebased upstream/master onto my working branch before opening this PR
- [ x ] I have named my PR something sensible
- [ x ] I have included the issue number above
- [ x ] I have assigned atleast two people to review my changes

## Other Notes
Other notes/context

### CODE REVIEW

File: backend/database.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const mysql = require('mysql');
+
+let connection = mysql.createConnection({
+    host: process.env.DB_HOST,
+    user: process.env.DB_USER,
+    database: process.env.DB_DATABASE,
+    password: process.env.DB_PASS
+});
+
+connection.connect(function (err) {
```
Comment (itsercleung) [inline backend/database.js]: Does the sql connection need to be closed at some point?

File: backend/routes/restaurant.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+require('dotenv').config()
+
+const bodyParser = require('body-parser');
+const connection = require('../database');
+
+const router = express.Router();
+router.use(bodyParser.urlencoded({ extended: true }));
+
+router.get('/', (req, res) => {
```
Comment (itsercleung) [inline backend/routes/restaurant.js]: Does the URI need to be explicit? "/getrestaurants", etc... ?

**Comment by itsercleung** [comment]
> Could you please include the new dependencies (dotenv and mysql) in the package.json so they can automatically be installed. Please also run the linter.

Is the package.json not updated?

**Comment by jacoballen4534** [comment]
> > Could you please include the new dependencies (dotenv and mysql) in the package.json so they can automatically be installed. Please also run the linter.
> 
> Is the package.json not updated?

Am I just looking at the wrong thing? In "files changed" it doesn't have package.json

**Comment by itsercleung** [comment]
> > > Could you please include the new dependencies (dotenv and mysql) in the package.json so they can automatically be installed. Please also run the linter.
> > 
> > 
> > Is the package.json not updated?
> 
> Am I just looking at the wrong thing? In "files changed" it doesn't have package.json

I see what you mean. Usually when you install packages in npm it should update the package.json. @wshin387 have you added the file in? Thanks :)

**Comment by jacoballen4534** [comment]
Would it be possible to move the database settings from .env to the config, then untrack the config.

**Comment by jacoballen4534** [comment]
Do we want to decide on how to handle errors? Atm it looks like it just crashes the server if you try to get a restaurant that doesn't exist.

**Comment by ljcnz** [comment]
> Do we want to decide on how to handle errors? Atm it looks like it just crashes the server if you try to get a restaurant that doesn't exist.

We could add this as a seperate issue? For now, I'll approve this merge. Thank you William for your work, and thank you jacob and eric for your reviews.

**Pull request merged**